src/main/java/org/sonatype/nexus/ci/iq/IqApplication.groovy

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.iq
+
+import com.thoughtworks.xstream.converters.basic.AbstractSingleValueConverter
+import hudson.model.Describable
+import hudson.model.Descriptor
+import hudson.util.FormValidation
+import jenkins.model.Jenkins
+import org.kohsuke.stapler.QueryParameter
+
+@SuppressWarnings('AbstractClassWithoutAbstractMethod')
+abstract class IqApplication
+    implements Describable<IqApplication>
+{
+  String applicationId
+
+  protected IqApplication(final String applicationId) {
+    this.applicationId = applicationId
+  }
+
+  @Override
+  Descriptor<IqApplication> getDescriptor() {
+    return Jenkins.getInstance().getDescriptorOrDie(this.getClass())
+  }
+
+  /**
+   * Provide Migration path from String to IqApplication
+   */
+  static final class ConverterImpl
```

Comment:
Is there a test we can write that exercises this?

Reply:
> Is there a test we can write that exercises this?

The 'it migrates a Freestyle IQ job' in ComToOrgMigratorIntegrationTest tests this code.  But maybe I should make an integration test just for this. Call it IqApplicationMigrationTest.groovy.  Is that a good idea?

Reply:
> The 'it migrates a Freestyle IQ job' in ComToOrgMigratorIntegrationTest tests this code. But maybe I should make an integration test just for this. Call it IqApplicationMigrationTest.groovy. Is that a good idea?

If the unit test has the coverage we are looking for, then we probably don't need an integration test.  Only need for an integration test would be if there are issues with the unit test not fully replicating real circumstances.

Reply:
[TRIVIAL] In my opinion there should be an integration test as you are solving an integration issue - the system has old XML that needs to be loaded into a new format.